src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -127,13 +162,23 @@ private async Task<OpenApiPaths> GetOpenApiPathsAsync(HashSet<OpenApiTag> captur
         {
             var operation = await GetOperationAsync(description, capturedTags, cancellationToken);
             operation.Extensions.Add(OpenApiConstants.DescriptionId, new OpenApiString(description.ActionDescriptor.Id));
-            _operationTransformerContextCache.TryAdd(description.ActionDescriptor.Id, new OpenApiOperationTransformerContext
+
+            var operationContext = new OpenApiOperationTransformerContext
             {
                 DocumentName = documentName,
                 Description = description,
                 ApplicationServices = serviceProvider,
-            });
+            };
+
+            _operationTransformerContextCache.TryAdd(description.ActionDescriptor.Id, operationContext);
             operations[description.GetOperationType()] = operation;
+
+            // Use index-based for loop to avoid allocating an enumerator with a foreach.
+            for (var i = 0; i < _options.OperationTransformers.Count; i++)
```

Comment:
I think this change violates the ordering principle for transformers that we originally established during API review ([ref](https://github.com/dotnet/aspnetcore/issues/54650#issuecomment-2023317760)) where operation transformers are modeled as document transformers and run in FIFO-order based on their registration. Now it seems like if an operation transformer is an activated instance, it'll run in order with document transformers but if it is a delegate or instantiated then it will run before document transformers.

Assuming my assesssment here is correct, we'll probably want to figure out a way to fix this.

Reply:
I've added a test for operations running FIFO before the document transformers and refactored things so it passes. The trade-off here was that it's one activation per operation now if the dependency is transient, rather than being created once per document.